src/main/java/ch/cern/eam/wshub/core/services/administration/impl/UserGroupMenuServiceImpl.java

```diff
@@ -0,0 +1,321 @@
+package ch.cern.eam.wshub.core.services.administration.impl;
+
+import ch.cern.eam.wshub.core.client.InforContext;
+import ch.cern.eam.wshub.core.services.administration.UserGroupMenuService;
+import ch.cern.eam.wshub.core.tools.ApplicationData;
+import ch.cern.eam.wshub.core.tools.InforException;
+import ch.cern.eam.wshub.core.tools.Tools;
+import net.datastream.schemas.mp_entities.extmenus_001.ExtMenus;
+import net.datastream.schemas.mp_entities.extmenushierarchy_001.ExtMenusHierarchy;
+import net.datastream.schemas.mp_fields.*;
+import net.datastream.schemas.mp_functions.mp6005_001.MP6005_GetExtMenusHierarchy_001;
+import net.datastream.schemas.mp_functions.mp6043_001.MP6043_AddExtMenus_001;
+import net.datastream.schemas.mp_functions.mp6045_001.MP6045_DeleteExtMenus_001;
+import net.datastream.wsdls.inforws.InforWebServicesPT;
+
+import java.util.*;
+
+public class UserGroupMenuServiceImpl implements UserGroupMenuService {
+    private Tools tools;
+    private InforWebServicesPT inforws;
+    private ApplicationData applicationData;
+
+    public UserGroupMenuServiceImpl(ApplicationData applicationData, Tools tools, InforWebServicesPT inforWebServicesToolkitClient) {
+        this.applicationData = applicationData;
+        this.tools = tools;
+        this.inforws = inforWebServicesToolkitClient;
+    }
+
+    private void addFolderToEntries(List<GenericMenuEntry> entries, FOLDER_Type parent) {
```

Comment:
Why is this variable called parent? Isn't it just a folder?

Reply:
That's a good question. But it no longer matters, method has been removed since implementation of tree data structure.